crates/compiler/builtins/roc/Json.roc

```diff
@@ -797,35 +806,36 @@ expect
 
 # JSON STRING PRIMITIVE --------------------------------------------------------
 
-# TODO add support for 'null' decoding
-
 # Decode a Json string primitive into a RocStr
 #
 # Note that decodeStr does not handle leading whitespace, any whitespace must be
 # handled in json list or record decodin.
 decodeString = Decode.custom \bytes, @Json {} ->
-    { taken: strBytes, rest } = bytes |> takeJsonString
-
-    if List.isEmpty strBytes then
-        { result: Err TooShort, rest: bytes }
+    if List.startsWith bytes ['n', 'u', 'l', 'l'] then
```

Comment:
Idea: I think the `when` pattern match version of this condition should run faster because it doesn't need to allocate the `['n', 'u', 'l', 'l']` list!